### PR TITLE
Feat/create docker container on push

### PR DESCRIPTION
**Pull request by Badmuts**
# Description

This PR adds new docker builds on push. Every push gets build by Github CI and tags the build with the branch name and commit sha. For example: `feat-create-docker-container-on-push-07a8c10`. This allows you to quickly test new features. On successful build this also creates a comment with the released image.

## Type of change

Feature

## Documentation

None

## Tests

Tested via GitHub workflow.

## Branch

`master`

### CODE REVIEW

File: .github/workflows/build-publish-docker.yml [hunk 1]
```diff
@@ -16,35 +18,52 @@ jobs:
...
+
+      # Comment build info in commit
+      - uses: actions/github-script@0.9.0
+        env:
+          STDOUT: "Published new image: `${{ fromJson(steps.docker_build.outputs.metadata)['image.name'] }}`"
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.repos.createCommitComment({
+              issue_number: context.issue.number,
```
Comment (rudivanhierden) [inline .github/workflows/build-publish-docker.yml:60]: `issue_number` isn't a parameter in `createCommitComment`, so I think it can be removed

**Review by rudivanhierden — COMMENTED** [review]
Looks good, I have a few questions but overall i'm happy with the results; the commit comments are a nice addition, and I like the use of `docker/metadata-action` for the tags 👍

File: .github/workflows/build-publish-docker.yml (hunk 1 above)
Comment (Badmuts) [reply .github/workflows/build-publish-docker.yml:60]: Yes this should be removed

File: .github/workflows/build-publish-docker.yml [hunk 2]
```diff
@@ -16,35 +18,52 @@ jobs:
...
-        id: get_version
-        run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
-
-   #   - name: tag number
-#        run : echo ${{ github.event.release.tag_name }}
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: openstad/api
```
[the remark this answers is not in the document]
Comment (Badmuts) [reply .github/workflows/build-publish-docker.yml]: This should now be configurable by setting a secret with the name `DOCKER_IMAGE`

**Pull request merged**